### PR TITLE
Labyrinth Anna Hasselström & Nathalie Nordström

### PR DESCRIPTION
**Pull request by nathalienordstrom**
https://labyrint-game.netlify.app

### CODE REVIEW

**Comment by nathalienordstrom** [comment]
a

**Review by Snabbina — CHANGES_REQUESTED** [review]
🤩 Hey, Anna and Nathalie!

What a mysterious game you've made!

You have worked well with the layout in the various media queries, and nice elements in the styling where there is a consistent theme in both color and shapes.

The main features are included which allows the game to go through from start to finish. and also the opportunity to go back a step or to start over is a practical feature.
Great job with the styled components! The naming of functions and variables is clear and consistent through out the code.

Also very impressed with your solution with the conditional rendering for the different components. 

We have a changes requested before approval though:
- Not consistent with ;, semikolon. For example in App.js and PlayingGame.js you use ; but in GameOver and loading you are not using them anymore. Change to one approach.
- indentation is missing in some places. 

Please see our comments, questions and extra appreciation for your code!
🎉

Best of the best from Sabina and Ellen!

**Comment by nathalienordstrom** [comment]
All done thank you Sabina and Ellen

File: code/src/components/StartingGame.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+              value={inputValue}
+              onChange={event => setInputValue(event.target.value)}
+            />
+          </Inputer>
+          <InputButton>
+            <Button onClick={updateUsername}>Save</Button>
+          </InputButton>
+        </InputContainer>
+        <ButtonContainer>
+          <Button type="button" onClick={() => dispatch(firstFetch(username))}>Play Game</Button>
```
[the remark this answers is not in the document]
Comment (HIPPIEKICK) [reply code/src/components/StartingGame.js]: I agree, the start button should be hidden or disabled if it's not possible to start the game 👍

File: code/src/components/GameOver.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        <>
+        <GameOverContainer>
+        <GameOverCard>
+        <GameOverText>{gameStatus.description}</GameOverText>
+          {/*This button will allow the user to go back to the starting page */}
+        <StartOverButton buttonwidth onClick={() => window.location.reload()}>Start New Game</StartOverButton>
+        </GameOverCard>
+        </GameOverContainer>
+        </>
+    )
```
[the remark this answers is not in the document]
Comment (HIPPIEKICK) [reply code/src/components/GameOver.js:19]: Yes, this should be indented. And no need for the fragment

File: code/src/components/Loading.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import styled, { keyframes } from "styled-components";
+
+export const Loading = () => {
+    const isLoading = useSelector((state) => state.ui.isLoading)
```
Comment (HIPPIEKICK) [inline code/src/components/Loading.js:8]: Great name of a boolean!

File: code/src/components/PlayingGame.js [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+const HeaderText = styled.h2`
+  font-size: 25px;
+  text-shadow: 2px 2px 4px #000000;
+  background: -webkit-linear-gradient(#eee, white);
+    -webkit-background-clip: text;
+    -webkit-text-fill-color: transparent;
+    @media (min-width: 768px) {
+      font-size: 32px;
+
+    }
```
[the remark this answers is not in the document]
Comment (HIPPIEKICK) [reply code/src/components/PlayingGame.js:99]: Goes for this whole file

File: code/src/reducers/currentstate.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    // Payload comes from the input value from StartingPage.
+    updateUsername: (state, action) => {
+      state.username = action.payload;
+    },
+
+    // A reducer that takes the user back to previous moves. It is initialzed as an empty array and
+    // it will be populated with the user moves.
+    // The reducer takes the currentGameState (where in game user is) and set that to the last move in the history array.
+    // It also removes, using slice(), the last move in the array, since that move becomes the current move when back-tracking.
+    historyGoBack: state => {
```
Comment (HIPPIEKICK) [inline code/src/reducers/currentstate.js:36]: Maybe `goBack` would be a fitting name?

File: code/src/reducers/currentstate.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { ui } from './ui'
```
Comment (HIPPIEKICK) [inline code/src/reducers/currentstate.js:2]: ;

File: code/src/reducers/ui.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { createSlice } from '@reduxjs/toolkit'
```
Comment (HIPPIEKICK) [inline code/src/reducers/ui.js:1]: Semicolons missing in this file ;)

**Comment by HIPPIEKICK** [comment]
Hi Anna and Nathalie! Please check indentation and semicolons in all files. Also, I would like to ask you to think about your starting screen. Right now it's not clear that you have to click Save before Play Game. Once you've saved - you get no indicator of that the save was successful. One suggestion is to only have the Save button visible to start with, and then after the save is done, that section is hidden and the start screen only shows the Play Game button. Alternatively, you could display a message to the user saying that the save was successful and apply some more styling to the disabled Play Game button so the user knows it's not possible to start.

**Pull request closed without merging**